core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -247,31 +318,37 @@ final void completeExceptionally(Throwable throwable, int topologyId) {
          future.completeExceptionally(throwable);
       }
 
-      abstract boolean hasPendingBackupAcks();
-
-      abstract void onMembersChange(Collection<Address> members);
-
       final boolean isWrongTopologyOrIsDone(int topologyId) {
          return this.topologyId != topologyId || future.isDone();
       }
    }
 
-   private class SingleKeyCollector extends BaseCollector<Object> {
+   private class SingleKeyCollector<T> extends BaseCollector<T> {
+      private final Address primaryOwner;
       private final Collection<Address> backupOwners;
+      private Collection<Address> unsolicitedAcks;
```

Comment:
Originator does not know who are the biased nodes before primary responds, but it can already receive the acks before the primary ack. So we keep the acks here and when primary ack arrives we remove them, not adding the backup owners.